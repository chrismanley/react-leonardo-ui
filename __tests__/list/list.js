import React from 'react';
import renderer from 'react-test-renderer';

import List from '../../src/list';

test('Empty list renders correctly', () => {
  const tree = renderer.create(
    <List />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('List with items renders correctly', () => {
  const tree = renderer.create(
    <List>
      <List.Header>Header</List.Header>
      <List.Item>Item 1</List.Item>
      <List.Item active>Item 1</List.Item>
      <List.Item disabled>Item 1</List.Item>
    </List>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('List with item custom content renders correctly', () => {
  const tree = renderer.create(
    <List>
      <List.Item>
        <List.Aside>Aside</List.Aside>
        <List.Text>Text</List.Text>
      </List.Item>
    </List>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('List passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <List className="my-class" data-x="x">
      <List.Header>Header</List.Header>
      <List.Item data-x="x" title="Item title" onClick={onClick}>Item</List.Item>
    </List>
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
