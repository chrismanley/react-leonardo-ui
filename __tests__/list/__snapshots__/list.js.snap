// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty list renders correctly 1`] = `
<ul
  className="lui-list"
/>
`;

exports[`List passes props correctly 1`] = `
<ul
  className="lui-list my-class"
  data-x="x"
>
  <li
    className="lui-list__header"
  >
    Header
  </li>
  <li
    className="lui-list__item"
    data-x="x"
    disabled={undefined}
    onClick={[Function]}
    title="Item title"
  >
    Item
  </li>
</ul>
`;

exports[`List with item custom content renders correctly 1`] = `
<ul
  className="lui-list"
>
  <li
    className="lui-list__item"
    disabled={undefined}
  >
    <div
      className="lui-list__aside"
    >
      Aside
    </div>
    <div
      className="lui-list__text"
    >
      Text
    </div>
  </li>
</ul>
`;

exports[`List with items renders correctly 1`] = `
<ul
  className="lui-list"
>
  <li
    className="lui-list__header"
  >
    Header
  </li>
  <li
    className="lui-list__item"
    disabled={undefined}
  >
    Item 1
  </li>
  <li
    className="lui-list__item lui-active"
    disabled={undefined}
  >
    Item 1
  </li>
  <li
    className="lui-list__item lui-disabled"
    disabled={true}
  >
    Item 1
  </li>
</ul>
`;
